bundle/jsky.app.ot/src/main/scala/jsky/app/ot/ags/BagsManager.scala

```diff
@@ -0,0 +1,321 @@
+package jsky.app.ot.ags
+
+import java.beans.{PropertyChangeEvent, PropertyChangeListener}
+import java.util.concurrent.{LinkedBlockingQueue, Executors}
+import java.util.logging.Logger
+
+import edu.gemini.ags.api.{AgsRegistrar, AgsStrategy}
+import edu.gemini.pot.sp._
+import edu.gemini.spModel.guide.GuideProbe
+import edu.gemini.spModel.obs.context.ObsContext
+import edu.gemini.spModel.rich.shared.immutable._
+import edu.gemini.spModel.target.env.{TargetEnvironment, GuideProbeTargets}
+import edu.gemini.spModel.target.obsComp.TargetObsComp
+import jsky.app.ot.OT
+import jsky.app.ot.gemini.altair.Altair_WFS_Feature
+import jsky.app.ot.gemini.inst.OIWFS_Feature
+import jsky.app.ot.gemini.tpe.TpePWFSFeature
+import jsky.app.ot.tpe.{GemsGuideStarWorker, GuideStarSupport, TpeContext, TpeManager}
+import scala.collection.JavaConverters._
+import scala.collection.immutable.HashMap
+import scala.swing.Swing
+import scala.util.{Try, Failure, Success}
+import scala.concurrent.ExecutionContext.Implicits.global
+
+
+object BagsManager {
```

Comment:
Okay, this is a much better idea and this was bothering me.
I created a new class to contain the code for non-GeMS strategies to apply the selections to observations and to update the TPE, as it being done in `GemsGuideStarWorker`, and it is much clearer and cleaner now, and the GeMS and non-GeMS cases in `BagsManager` are now much more similar, as they should be.

Thanks. I am much happier with the state of affairs now, so I appreciate that.